City-of-Helsinki/drupal-asuntotuotanto
# SATT-70: Clean application applicant information

### Description

Clean applicant information on applications


### How to test it

- use dev db and be sure that you have ASU_DJANGO_BACKEND_URL & DRUPAL_AUTH_TOKEN setup
- make fresh
- check application data on sql

   a) make shell && drush sql-cli
     - Check non saved application(field_locked = 0) still have applicant information
       - `select * from asu_application where field_locked = 0 order by id desc limit 5;`
     - Check saved application(field_locked = 1) doesn't have applicant information
       - `select * from asu_application where field_locked = 1 order by id desc limit 5;`
       
   b) open sql with sequel
     - Check non saved application(field_locked = 0) still have applicant information
     - Check saved application(field_locked = 1) doesn't have applicant information
     
- login as admin
- rebuild track information and re-index both search api indexes https://asuntotuotanto.docker.so/fi/admin/config/search/search-api
- go haso listing page
- be sure that you have 2 projects where you can apply application (like /application/add/haso/xxx)
   - project State of sale is **myynnissä** and in aikataulut tab **Hakuajan päättymisaika** is **future** date
   - after you have save node(s) you can check search api statuses and clear cache
- now go users listing page and check some salesperson uid
- logout as admin and login as saleperson
- go user listing page
- create application for tero ayramo user on actions "Luo hakemus uudelle käyttäjälle"
- now on project listing select some active project which lead to you on /application/add/haso/xxx path and form **Not** apply for for free apply_for_free_apartment?apartment=XX&project=X
- fill form and use correct testing birth date 01.01.70 and personal id 999R
- save form and if everything goes without errors then go check sql doesn't have applicant information
- make shell && drush sql-cli
- `select * from asu_application order by id desc limit 1;`
- logout as salesperson and login as Tero uid 105 using drush uli
- Now go haso listing page and do application to another project not that same one which you use on salesperson
- application path is /application/add/haso/xxx and **not** apply_for_free_apartment?apartment=XX&project=X
- fill application by correct testing birth date 01.01.70 & personal id 999R and **save that on draft** (tallenna luonnoksena)
- you can now check that applicant data is on application on sql
  - make shell && drush sql-cli
  - `select * from asu_application order by id desc limit 1;`
- Now you can open application edit and save (sent) application
- now check applicant information is cleared on sql 
  - make shell && drush sql-cli
  - `select * from asu_application order by id desc limit 1;`
